### ISSUE TITLE
Invalid Bundle Structure - for including FirebaseABTesting & FirebaseRemoteConfig frameworks

### ISSUE THREAD

**Issue by joonfromla** [distriqt/ANE-Firebase#186]
I was uploading my release ipa to Appstore using the Xcode Application Loader, when I received the following errors:

```
ERROR ITMS-90171: "Invalid Bundle Structure - The binary file 'fdtFeathEchoApp.app/FirebaseABTesting.framework/FirebaseABTesting' is not permitted. Your app can’t contain standalone executables or libraries, other than a valid CFBundleExecutable of supported bundles. Refer to the Bundle Programming Guide at https://developer.apple.com/go/?id=bundle-structure for information on the iOS app bundle structure."

ERROR ITMS-90171: "Invalid Bundle Structure - The binary file 'fdtFeathEchoApp.app/FirebaseRemoteConfig.framework/FirebaseRemoteConfig' is not permitted. Your app can’t contain standalone executables or libraries, other than a valid CFBundleExecutable of supported bundles. Refer to the Bundle Programming Guide at https://developer.apple.com/go/?id=bundle-structure for information on the iOS app bundle structure."
```

I have tried using the 3 latest versions of the Firebase ANEs, but I still receive the same errors.

Firebase Versions tried:
2.4.070
2.4
2.3

Here are all the ANE I'm including in my project:
```
    com.distriqt.androidsupport.CustomTabs
    com.distriqt.MediaPlayer
    com.distriqt.androidsupport.CardViewV7
    com.distriqt.androidsupport.V4
    com.distriqt.Message
    com.distriqt.Bolts
    com.distriqt.firebase.Crashlytics
    com.fabricemontfort.air.ezSpeech
    com.distriqt.androidsupport.AppCompatV7
    com.distriqt.androidsupport.Design
    com.distriqt.CustomResources
    com.distriqt.Core
    com.distriqt.playservices.Auth
    com.distriqt.AudioRecorder
    com.distriqt.Share
    com.distriqt.androidsupport.RecyclerViewV7
    com.distriqt.firebase.Performance
    com.distriqt.Application
    com.distriqt.playservices.Base
    com.distriqt.Firebase
    com.distriqt.FacebookAPI
    com.distriqt.firebase.RemoteConfig
```

- AIR SDK version: 31
- Device OS: ios
- Development IDE: FDT

My error seems similar to issue #35, but I didn't see any suggestions there helps my situation.

**Comment by joonfromla** [comment]
I discovered that the error messages disappeared after I remove these two ANEs:
com.distriqt.firebase.Performance
com.distriqt.firebase.RemoteConfig

I also see that both those ANEs contain these two libraries:
FirebaseABTesting.framework
FirebaseRemoteConfig.framework

Maybe removing these libraries from one or the other ANE might avoid the error messages?

**Comment by marchbold** [comment]
Thanks for reporting this issue. Could you try the following build for us to confirm the fix?

https://www.dropbox.com/s/uxzzej8xfp41s5n/com.distriqt.Firebase.v2.4.071.zip?dl=0

**Comment by joonfromla** [comment]
Thanks, I'll implement this new version and let  you know my results tomorrow.

**Comment by joonfromla** [comment]
I published my ios app with the Firebase ANE 2.4.071 you provided.  Specifically, I used Performance.ane and RemoteConfig.ane, and I was able to submit my IPA the the Appstore without encountering the two ERROR ITMS-90171 errors.  Hurray!!!  Of course, it takes a day to see the Firebase Performance logs.  I will let you know if I encounter any logs that indicates Performance Monitoring is not working as usual.  For now, your fix has removed this bug.  Thanks for your help!!
